src/MonoTorrent/MonoTorrent.Client/Managers/InactivePeerManager.cs

```diff
@@ -121,8 +131,8 @@ internal void TimePassed ()
 
             // We've found a peer to disconnect
             // Add it to the inactive list for this torrent and disconnect it
-            InactivePeerList.Add (TorrentManager.Peers.ConnectedPeers[peerToDisconnect].Uri);
-            ConnectionManager.CleanupSocket (TorrentManager, TorrentManager.Peers.ConnectedPeers[peerToDisconnect]);
```

Comment:
Just to make sure my understanding is correct - this is executing *after* we've finished looping, so we're not mutating the `ConnectedPeers` list while we're iterating over it. Right?